File: main_routes/models/books.js
const mongoose = require('mongoose');

const BookSchema = mongoose.Schema({
    title: String,
    authors: String,
    average_rating: Number,
    isbn: String,
    isbn13: String,
    language_code: String,
    num_pages: Number,
    ratings_count: Number,
    text_reviews_count: Number,
    publication_date: String,
    publisher: String,
    quantity_total: {type: Number, default: 1},
    quantity_in_library: {type: Number, default: 1},
    rent: {type: Number, default: 50},
    times_borrowed: {type: Number, default: 0},
    added_at :{
        type:Date,
        default:new Date()
    }
});

const Book = mongoose.model('book', BookSchema);

module.exports = Book;

File: flask-backend/models/books.py
from ..database import *
from datetime import datetime

db = get_db()

class book(db.Document):
    title = db.StringField(required=True)
    authors = db.StringField(required=True)
    isbn = db.StringField(required=True)
    isbn13 = db.StringField(required=True)
    language_code = db.StringField()
    num_pages = db.StringField()
    ratings_count = db.StringField()
    text_reviews_count = db.StringField()
    publication_date = db.DateTimeField()
    publisher = db.StringField()
    quantity_total = db.IntegerField(default=1)
    quantity_in_library =db.IntegerField(default=1)
    rent = db.IntegerField(default=50)
    times_borrowed = db.IntegerField(default=0)
    created_at = db.DateTimeField(default=datetime.utcnow)
    






